python/cudf/cudf/dataframe/categorical.py

```diff
@@ -45,15 +45,114 @@ def codes(self):
         else:
             return Series(data)
```

Comment:
```suggestion
            return Series(data, index=self._parent.index, name=self._parent.name)
```

Reply:
The above Series constructor that handles masked columns should handle indexes and names as well.